src/simulations/crmp/clair_data.py

```diff
@@ -0,0 +1,172 @@
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+
+from sklearn.model_selection import train_test_split
+
+from src.config import INPUTS
+from src.data.read_clair import (
+    injectors, producers, producer_names, producer_starting_indicies, time
+)
+from src.helpers.analysis import fit_statistics
+from src.helpers.features import production_rate_dataset, producer_rows_from_df
+from src.helpers.models import model_namer
+from src.models.crmp import CRMP
+
+
+fit_ouput_file = INPUTS['real_data']['fit']['sensitivity_analysis']
+predict_output_file = INPUTS['real_data']['predict']['sensitivity_analysis']
+fit_data = {
+    'Producer': [], 'Model': [], 'tau_initial': [], 'tau_final': [],
+    'f1_initial': [], 'f1_final': [], 'f2_initial': [], 'f2_final': [],
+    'f3_initial': [], 'f3_final': [], 'f4_initial': [], 'f4_final': [],
+    'r2': [], 'MSE': []
+}
+predict_data = {
+    'Producer': [], 'Model': [], 'tau_initial': [], 'tau_final': [],
+    'f1_initial': [], 'f1_final': [], 'f2_initial': [], 'f2_final': [],
+    'f3_initial': [], 'f3_final': [], 'f4_initial': [], 'f4_final': [],
+    'r2': [], 'MSE': []
+}
+
+
+number_of_gains = 6
+number_of_time_constants = 11
+gains = np.linspace(0, 1, number_of_gains)
+tau = np.linspace(0, 100, number_of_time_constants)
+p0s = []
+for i in tau:
+    if i == 0:
+        i = 1e-06
+    for j in gains:
+        for k in gains:
+            for l in gains:
+                for m in gains:
+                    if (j + k + l + m) != 1:
+                        continue
+                    else:
+                        p0s.append((i, j, k, l, m))
```

Comment:
There has to be a better way to do this. Make an issue with to address this. Google "constrained permutation"